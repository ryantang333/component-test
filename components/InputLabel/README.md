# @test/input-label

Welcome to my `InputLabel` component.

## Installation

```sh
npm i @test/input-label
# or with yarn
yarn add @test/input-label
```

Then to use the component in your code just import it!

```js
import InputLabel from '@test/input-label';
// and with css-modules
import '@test/input-label/dist/main.css';
```
