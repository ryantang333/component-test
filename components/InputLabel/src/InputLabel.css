.input-label {
  color: rgba(0, 0, 0, 0.6);
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375rem;
  padding: 0 0.625rem;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  position: absolute;
  background: #fff;
  left: 0;
  top: 0;
  transform: translate(0.875rem, -0.5625rem) scale(0.75);
  transition: color 200ms cubic-bezier(0, 0, 0.2, 1) 0ms,
    transform 200ms cubic-bezier(0, 0, 0.2, 1) 0ms,
    max-width 200ms cubic-bezier(0, 0, 0.2, 1) 0ms;
}

.input-label.error {
  color: #d32f2f;
}
