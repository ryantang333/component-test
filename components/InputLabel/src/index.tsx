import React from 'react';
import cn from 'classnames';
import styles from './InputLabel.css';

interface InputLabelProps {
  /** Error boolean to change styles */
  error?: boolean;
  /** ID for corresponding input */
  htmlFor?: string;
  /** Input label */
  label?: string;
  /** User-provided styles */
  style?: React.CSSProperties;
}

/** Label component that accompanies input elements. */
export const InputLabel = (props: InputLabelProps) => {
  const { error, htmlFor, label, style } = props;
  const labelClasses = cn(styles.inputLabel, { [styles.error]: error });

  return (
    <label className={labelClasses} htmlFor={htmlFor} style={style}>
      {label}
    </label>
  );
};

export default InputLabel;
