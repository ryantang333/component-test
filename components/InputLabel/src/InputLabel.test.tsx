import React from 'react';
import { render } from '@testing-library/react';
import InputLabel from '.';

describe('InputLabel', () => {
  test('It matches the snapshot', () => {
    const { container } = render(<InputLabel />);
    expect(container).toMatchSnapshot();
  });
});
