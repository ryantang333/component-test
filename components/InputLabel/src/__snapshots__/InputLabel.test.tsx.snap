// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel It matches the snapshot 1`] = `
<div>
  <label
    class="inputLabel"
  />
</div>
`;
